src/storage/src/render/sinks.rs

```diff
@@ -66,34 +59,24 @@ pub(crate) fn render_sink<G: Scope<Timestamp = Timestamp>>(
         // Copy the logic in DeltaJoin/Get/Join to start.
         |_timer, count| count > 1_000_000,
     );
-    needed_tokens.push(source_token);
+    tokens.push(source_token);
 
     let ok_collection =
         apply_sink_envelope(sink_id, sink, &sink_render, ok_collection.as_collection());
 
-    let healthchecker_args = HealthcheckerArgs {
-        persist_clients: Arc::clone(&storage_state.persist_clients),
-        persist_location: sink.from_storage_metadata.persist_location.clone(),
-        status_shard_id: sink.status_id,
-        now_fn: storage_state.now.clone(),
-    };
-
-    let sink_token = sink_render.render_continuous_sink(
+    let (health, sink_token) = sink_render.render_continuous_sink(
         storage_state,
         sink,
         sink_id,
         ok_collection,
         err_collection.as_collection(),
-        healthchecker_args,
     );
 
     if let Some(sink_token) = sink_token {
-        needed_tokens.push(sink_token);
+        tokens.push(sink_token);
     }
 
-    storage_state
-        .sink_tokens
-        .insert(sink_id, SinkToken::new(Box::new(needed_tokens)));
+    health.map(|status| (0, status)).leave()
```

Comment:
I feel like it's weird that we're leaving but not entering in here. Could we move the leave outside and simplify the signature?

Reply:
This is the same way we do it in sources so i'd love to leave it like this